warble/functions.cc

```diff
@@ -156,19 +164,40 @@ bool Profile(Database* db, Any req, Any* rep) {
       db->get("_followers_" + username);
   if (followers) {
     for (std::string f : *followers) {
-      std::string* s = reply.add_followers();
+      std::string *s = reply.add_followers();
       *s = f;
     }
   }
   std::optional<std::vector<std::string>> following =
       db->get("_following_" + username);
   if (following) {
     for (std::string f : *following) {
-      std::string* s = reply.add_following();
+      std::string *s = reply.add_following();
       *s = f;
     }
   }
   rep->PackFrom(reply);
   LOG(INFO) << "Got profile for user " << username;
   return true;
+}
+
+bool Stream(Database *db, Any req, Any *rep) {
+  StreamRequest request;
+  StreamReply reply;
+  req.UnpackTo(&request);
+  std::string hashtag = request.hashtag();
+  std::optional<std::vector<std::string>> exists =
+      db->get("_hashtag_" + hashtag);
```

Comment:
not 100% sure if this is a bug, but I think this might not work in the case when the client has initiated streaming of `#hash` after someone has previously posted a Warble already containing `#hash`. currently the old Warble would be sent out to the newer streaming request when it should not be, but double check me on this

Reply:
In cli.cc, I do an initial call to function.cc's Stream() and get the size of the vector. Then, in cli.cc's while(1) loop, the inner for loop starts at the size of the vector. This ensures that older hashtags are ignored. Again, pretty inefficient, but easy workaround.

Reply:
ah okay I did not see the initial request. then yes, it works